gdnative-core/src/core_types/geom/transform.rs

```diff
@@ -28,11 +35,80 @@ impl Transform {
         unsafe { std::mem::transmute::<sys::godot_transform, Self>(c) }
     }
 
+    /// Creates a `Transform` with no scaling or rotation at `origin`.
     #[inline]
     pub fn translate(origin: Vector3) -> Transform {
         Transform {
             basis: Basis::identity(),
             origin,
         }
     }
+
+    #[inline]
+    pub const fn identity() -> Transform {
+        Transform {
+            basis: Basis::identity(),
+            origin: Vector3::ZERO,
+        }
+    }
+
+    /// Returns a vector transformed (multiplied) by the matrix.
+    #[inline]
+    pub fn xform(&self, v: Vector3) -> Vector3 {
+        self.basis.xform(v) + self.origin
+    }
+
+    /// Returns a vector transformed (multiplied) by the transposed basis
+    /// matrix.
+    ///
+    /// **Note:** This results in a multiplication by the inverse of the matrix
+    /// only if it represents a rotation-reflection.
+    #[inline]
+    pub fn xform_inv(&self, v: Vector3) -> Vector3 {
+        self.basis.xform_inv(v - self.origin)
+    }
+
+    /// Returns the inverse of the transform, under the assumption that the
+    /// transformation is composed of rotation and translation (no scaling, use
+    /// affine_inverse for transforms with scaling).
+    #[inline]
+    pub fn inverse(&self) -> Transform {
+        let basis_inv = self.basis.transposed();
+        let origin_inv = basis_inv.xform(-self.origin);
+        Transform {
+            origin: origin_inv,
+            basis: basis_inv,
+        }
+    }
+
+    /// Returns the inverse of the transform, under the assumption that the
+    /// transformation is composed of rotation, scaling and translation.
+    #[inline]
+    pub fn affine_inverse(&self) -> Transform {
+        let basis_inv = self.basis.inverted();
+        let origin_inv = basis_inv.xform(-self.origin);
+        Transform {
+            origin: origin_inv,
+            basis: basis_inv,
+        }
+    }
+
+    /// Returns a copy of the transform rotated such that its -Z axis points
+    /// towards the target position.
+    ///
+    /// The transform will first be rotated around the given up vector, and then
+    /// fully aligned to the target by a further rotation around an axis
+    /// perpendicular to both the target and up vectors.
+    #[inline]
+    pub fn looking_at(eye: Vector3, target: Vector3, up: Vector3) -> Transform {
```

Comment:
Same decision would also affect the already present `translate(origin)` vs. `translated(&self, origin)`.

Reply:
> Any particular reason why this deviates from the GDScript method signature?

The main reason is that I copied this from a utility method in my code that used these arguments. But I think it makes more sense to make it consistent with the API :+1: 

> Same decision would also affect the already present translate(origin) vs. translated(&self, origin)

Indeed. I didn't want to change that because it was already there so it would become a breaking change. But since you're mentioning it I went ahead and created the following methods:

- `from_position`, renamed from the old `translate`.
- `translate`, which takes `&mut self` and modifies in-place
- `translated`, which takes `&self` and returns a modified transform

I think this is closer to the other APIs.

Reply:
> translated, which takes &self and returns a modified transform

This one definitely makes sense.

I'm not 100% sure about the others -- I would tend to go for the minimal API (close to Godot). But if we have `translate()`, we would also need `rotate()` and `scale()`, and do the same for `Transform2D`. Same with `from_position()`, which can currently be expressed as `IDENTITY.translated(v)`.

I'm not categorically against convenience functions, but for now I'd stay with the building blocks and encourage the "chain different transforms" pattern.